<bug id='2799' author='kranthik13' open_date='2020-12-22T12:22:09Z' closed_time='2021-01-02T22:28:18Z'>
	<summary>Unable to run ASR : JSONDecodeError</summary>
	<description>
I was testing the ASR pretrained model tedlium2.transformer.v1 on Google Colab using my sample wav file which had frame rate of 16000 and 1 channel.
Basic environments:

OS information: Linux 4.19.112+ #1 SMP Thu Jul 23 08:00:38 PDT 2020 x86_64
python version: 3.6.9 (default, Oct  8 2020, 12:12:24)  [GCC 8.4.0]
espnet version: espnet 0.9.6
Git hash [53f6aa1]

Commit date [Tue Dec 22 04:34:09 2020 +0000]


pytorch version: 1.1.0

Environments from torch.utils.collect_env:
&lt;denchmark-code&gt;Warning! You haven't set Python environment yet. Go to /content/espnet/tools and generate 'activate_python.sh'
Collecting environment information...
PyTorch version: 1.1.0
Is debug build: No
CUDA used to build PyTorch: 9.0.176

OS: Ubuntu 18.04.5 LTS
GCC version: (Ubuntu 7.5.0-3ubuntu1~18.04) 7.5.0
CMake version: version 3.12.0

Python version: 3.6
Is CUDA available: Yes
CUDA runtime version: 10.1.243
GPU models and configuration: GPU 0: Tesla P100-PCIE-16GB
Nvidia driver version: 418.67
cuDNN version: /usr/lib/x86_64-linux-gnu/libcudnn.so.7.6.5

Versions of relevant libraries:
[pip3] numpy==1.19.4
[pip3] pytorch-ranger==0.1.1
[pip3] pytorch-wpe==0.0.0
[pip3] torch==1.1.0
[pip3] torch-complex==0.2.0
[pip3] torch-optimizer==0.0.1a17
[pip3] torchsummary==1.5.1
[pip3] torchtext==0.3.1
[pip3] torchvision==0.8.1+cu101
[pip3] warpctc-pytorch==0.1.1
[conda] Could not collect
&lt;/denchmark-code&gt;

Task information:

Task: ASR
Recipe: telium2

Execution command:
&lt;denchmark-code&gt;cd /content/espnet/egs/tedlium2/asr1; bash ../../../utils/recog_wav.sh --models tedlium2.transformer.v1  /content/naruto_1.wav
&lt;/denchmark-code&gt;

Output for the above command is :
&lt;denchmark-code&gt;Warning! You haven't set Python environment yet. Go to /content/espnet/tools and generate 'activate_python.sh'
stage 0: Data preparation
stage 1: Feature Generation
steps/make_fbank_pitch.sh --cmd run.pl --nj 1 --write_utt2num_frames true decode/naruto_1/data decode/naruto_1/log decode/naruto_1/fbank
Warning! You haven't set Python environment yet. Go to /content/espnet/tools and generate 'activate_python.sh'
steps/make_fbank_pitch.sh: moving decode/naruto_1/data/feats.scp to decode/naruto_1/data/.backup
utils/validate_data_dir.sh: WARNING: you have only one speaker.  This probably a bad idea.
   Search for the word 'bold' in http://kaldi-asr.org/doc/data_prep.html
   for more information.
utils/validate_data_dir.sh: Successfully validated data-directory decode/naruto_1/data
steps/make_fbank_pitch.sh: [info]: no segments file exists: assuming wav.scp indexed by utterance.
steps/make_fbank_pitch.sh: Succeeded creating filterbank and pitch features for data
/content/espnet/egs/tedlium2/asr1/../../../utils/dump.sh --cmd run.pl --nj 1 --do_delta false decode/naruto_1/data/feats.scp decode/download/tedlium2.transformer.v1/data/train_trim_sp/cmvn.ark decode/naruto_1/log decode/naruto_1/dump
Warning! You haven't set Python environment yet. Go to /content/espnet/tools and generate 'activate_python.sh'
stage 2: Json Data Preparation
/content/espnet/egs/tedlium2/asr1/../../../utils/data2json.sh --feat decode/naruto_1/dump/feats.scp decode/naruto_1/data decode/naruto_1/dict
/content/espnet/egs/tedlium2/asr1/../../../utils/feat_to_shape.sh --cmd run.pl --nj 1 --filetype  --preprocess-conf  --verbose 0 decode/naruto_1/dump/feats.scp decode/naruto_1/data/tmp-3ZOjA/input_1/shape.scp
sym2int.pl: replacing X with 1
** Replaced 1 instances of OOVs with 1
stage 3: Decoding
run.pl: job failed, log is in decode/naruto_1/log/decode.log
&lt;/denchmark-code&gt;

Error Logs:
Output Error Log (decode/naruto_1/log/decode.log)  :
&lt;denchmark-code&gt;# asr_recog.py --config decode/download/tedlium2.transformer.v1/conf/decode.yaml --ngpu 0 --backend pytorch --debugmode 1 --verbose 1 --recog-json decode/naruto_1/dump/data.json --result-label decode/naruto_1/result.json --model decode/download/tedlium2.transformer.v1/exp/train_trim_sp_pytorch_nbpe500_ngpu2_train_pytorch_transformer.v2_epochs100_specaug/results/model.last10.avg.best.ep86_irielmep1_beam40 --api v2 --rnnlm decode/download/tedlium2.transformer.v1/exp/train_rnnlm_pytorch_lm_irie_batchsize128_unigram500/rnnlm.model.best 
# Started at Tue Dec 22 11:57:04 UTC 2020
#
2020-12-22 11:57:05,112 (asr_recog:305) INFO: python path = /env/python
2020-12-22 11:57:05,112 (asr_recog:310) INFO: set random seed = 1
2020-12-22 11:57:05,112 (asr_recog:321) INFO: backend = pytorch
2020-12-22 11:57:07,272 (recog:34) WARNING: experimental API for custom LMs is selected by --api v2
2020-12-22 11:57:07,273 (deterministic_utils:26) INFO: torch type check is disabled
2020-12-22 11:57:07,273 (asr_utils:656) INFO: reading a config file from decode/download/tedlium2.transformer.v1/exp/train_trim_sp_pytorch_nbpe500_ngpu2_train_pytorch_transformer.v2_epochs100_specaug/results/model.json
2020-12-22 11:57:07,273 (asr_init:169) WARNING: reading model parameters from decode/download/tedlium2.transformer.v1/exp/train_trim_sp_pytorch_nbpe500_ngpu2_train_pytorch_transformer.v2_epochs100_specaug/results/model.last10.avg.best.ep86_irielmep1_beam40
2020-12-22 11:57:07,430 (fill_missing_args:42) INFO: attribute "transformer_encoder_selfattn_layer_type" does not exist. use default selfattn.
2020-12-22 11:57:07,430 (fill_missing_args:42) INFO: attribute "transformer_decoder_selfattn_layer_type" does not exist. use default selfattn.
2020-12-22 11:57:07,430 (fill_missing_args:42) INFO: attribute "wshare" does not exist. use default 4.
2020-12-22 11:57:07,430 (fill_missing_args:42) INFO: attribute "ldconv_encoder_kernel_length" does not exist. use default 21_23_25_27_29_31_33_35_37_39_41_43.
2020-12-22 11:57:07,431 (fill_missing_args:42) INFO: attribute "ldconv_decoder_kernel_length" does not exist. use default 11_13_15_17_19_21.
2020-12-22 11:57:07,431 (fill_missing_args:42) INFO: attribute "ldconv_usebias" does not exist. use default False.
2020-12-22 11:57:07,462 (encoder:148) INFO: encoder self-attention layer type = self-attention
2020-12-22 11:57:07,641 (decoder:125) INFO: decoder self-attention layer type = self-attention
2020-12-22 11:57:07,885 (io_utils:59) WARNING: [Experimental feature] Some preprocessing will be done for the mini-batch creation using Transformation(
    0: TimeWarp(max_time_warp=5, inplace=True, mode=PIL)
    1: FreqMask(F=30, n_mask=2, replace_with_zero=False, inplace=True)
    2: TimeMask(T=40, n_mask=2, replace_with_zero=False, inplace=True))
2020-12-22 11:57:07,885 (asr_utils:656) INFO: reading a config file from decode/download/tedlium2.transformer.v1/exp/train_rnnlm_pytorch_lm_irie_batchsize128_unigram500/model.json
2020-12-22 11:57:10,939 (recog:123) INFO: Decoding device=cpu, dtype=torch.float32
Traceback (most recent call last):
  File "/content/espnet/egs/tedlium2/asr1/../../../espnet/bin/asr_recog.py", line 363, in &lt;module&gt;
    main(sys.argv[1:])
  File "/content/espnet/egs/tedlium2/asr1/../../../espnet/bin/asr_recog.py", line 333, in main
    recog_v2(args)
  File "/content/espnet/espnet/asr/pytorch_backend/recog.py", line 129, in recog_v2
    js = json.load(f)["utts"]
  File "/usr/lib/python3.6/json/__init__.py", line 299, in load
    parse_constant=parse_constant, object_pairs_hook=object_pairs_hook, **kw)
  File "/usr/lib/python3.6/json/__init__.py", line 354, in loads
    return _default_decoder.decode(s)
  File "/usr/lib/python3.6/json/decoder.py", line 339, in decode
    obj, end = self.raw_decode(s, idx=_w(s, 0).end())
  File "/usr/lib/python3.6/json/decoder.py", line 357, in raw_decode
    raise JSONDecodeError("Expecting value", s, err.value) from None
json.decoder.JSONDecodeError: Expecting value: line 1 column 1 (char 0)
# Accounting: time=7 threads=1
# Ended (code 1) at Tue Dec 22 11:57:11 UTC 2020, elapsed time 7 seconds
&lt;/denchmark-code&gt;

	</description>
	<comments>
		<comment id='1' author='kranthik13' date='2020-12-28T14:26:17Z'>
		I met the same error when I was playing around with the TTS demo (&lt;denchmark-link:https://github.com/espnet/interspeech2019-tutorial&gt;https://github.com/espnet/interspeech2019-tutorial&lt;/denchmark-link&gt;
).
Looking at JSON files generated, you will see the Warning! You haven't set Python environment yet. message at the top, hence JSONDecodeError.
A workaround is to edit /content/espnet/tools/activate_python.sh and comment the echo command out.
		</comment>
		<comment id='2' author='kranthik13' date='2021-01-02T21:12:04Z'>
		thanks, this should be fixed though it doesn't happen if you normally followed our installation.
		</comment>
	</comments>
</bug>